src/Nodejs.h

```diff
@@ -21,16 +21,21 @@ namespace plugin::Nodejs {
 
 class EventHandler;
 
+struct InitOptions {
+  std::string main_script_source;
+  std::vector<std::filesystem::path> files;
+  size_t initial_heap_size_in_bytes;
+  size_t maximum_heap_size_in_bytes;
+  bool exit_on_uncaught_exceptions;
```

Comment:
Are you omitting initializers for fields of basic types on purpose? IMO encoding defaults here and never leaving them uninitialized would be a good idea and of negligible overhead (using uninitialized variables triggers UB in C++), e.g.,

```
  bool exit_on_uncaught_exceptions = false;
```